### PR TITLE
add  Frame End Call Back and Animation Clip End Call Back

### PR DESCRIPTION
**Pull request by geron-cn**
add  Frame End Call Back and Animation Clip End Call Back

### CODE REVIEW

**Comment by geron-cn** [comment]
@VirtualSJ  @pandamicro @jun-bing could you check this out, thx.

File: cocos/editor-support/cocostudio/ActionTimeline/CCActionTimeline.cpp [hunk 1]
```diff
@@ -355,6 +373,52 @@ void ActionTimeline::emitFrameEvent(Frame* frame)
...
 
+void ActionTimeline::addFrameEndCallFunc(int frameIndex, const std::string& funcKey, std::function<void()> func)
+{
+    if (func != nullptr)
+    {
+        _frameEndCallFuncs[frameIndex][funcKey] = func;
+    }
+}
+
+void ActionTimeline::removeFrameEndCall(int frameIndex, const std::string& funcKey)
```
Comment (pandamicro) [inline cocos/editor-support/cocostudio/ActionTimeline/CCActionTimeline.cpp:384]: Please keep API name consistency: rename `ActionTimeline::removeFrameEndCall` to `ActionTimeline::removeFrameEndCallFunc`

File: cocos/editor-support/cocostudio/ActionTimeline/CCActionTimeline.cpp [hunk 2]
```diff
@@ -355,6 +373,52 @@ void ActionTimeline::emitFrameEvent(Frame* frame)
...
+    {
+        auto funcIter = (*endClipCallsIter).second.find(funcKey);
+        if (funcIter != (*endClipCallsIter).second.end())
+            (*endClipCallsIter).second.erase(funcKey);
+        if ((*endClipCallsIter).second.empty())
+            _frameEndCallFuncs.erase(endClipCallsIter);
+    }
+}
+
+void ActionTimeline::removeFrameEndCall(int frameIndex)
```
Comment (pandamicro) [inline cocos/editor-support/cocostudio/ActionTimeline/CCActionTimeline.cpp:397]: Please keep API name consistency: rename `ActionTimeline:: removeFrameEndCall` to `ActionTimeline::removeFrameEndCallFunc`

File: cocos/editor-support/cocostudio/ActionTimeline/CCActionTimeline.cpp [hunk 3]
```diff
@@ -355,6 +373,52 @@ void ActionTimeline::emitFrameEvent(Frame* frame)
...
+{
+    auto endClipCallsIter = _frameEndCallFuncs.find(frameIndex);
+    if (endClipCallsIter != _frameEndCallFuncs.end())
+    {
+        _frameEndCallFuncs.erase(endClipCallsIter);
+    }
+}
+
+void ActionTimeline::clearFrameEndCalls()
```
Comment (pandamicro) [inline cocos/editor-support/cocostudio/ActionTimeline/CCActionTimeline.cpp:406]: Please keep API name consistency: rename `ActionTimeline::clearFrameEndCalls` to `ActionTimeline::clearFrameEndCallFuncs`

**Comment by pandamicro** [comment]
Sorry I merged this PR by accident, it still remains some problems in added APIs, please add another PR to fix it, @geron-cn Thanks.

**Pull request merged**